### ISSUE TITLE
[Feature Request] LightGBMTunerCV model object

### ISSUE THREAD

**Issue by rohan-gt** [optuna/optuna#1740]
Is there a way to get the LightGBMTunerCV model object out for prediction purposes? This is because I'm using `n_estimators=10000` and `early_stopping=100` and when I use `best_params` I don't get the `n_estimators` value where lightgbm.cv() stopped at and I'll have to end up using `n_estimators=10000` itself to retrain on the entire dataset since I don't want to further split it into train and eval

**Comment by toshihikoyanase** [comment]
@rohan-gt Thank you for your feature request. 

> Is there a way to get the LightGBMTunerCV model object out for prediction purposes?

Actually, @nyanhi is working on it in #1702 . It is under review, but you can try it using his branch.

**Comment by rohan-gt** [comment]
@toshihikoyanase how do I do a pop install from his branch?

**Comment by toshihikoyanase** [comment]
You can install it via the following command:

```console
pip install --quiet git+https://github.com/nyanhi/optuna@feature/return-lgbmtunercv-booster
```

In addition, you need `lightgbm>=3.0.0` to use the feature:

```console
pip install --pre -U lightgbm
```

I created a [notebook](https://colab.research.google.com/drive/1mxnqVcUKA4NZHJj29GUxPJmEMhEK-rZx?usp=sharing), so please take a look.

**Comment by rohan-gt** [comment]
@toshihikoyanase thanks!

**Comment by rohan-gt** [comment]
@toshihikoyanase the returned model outputs 3 sets of predictions for the 3 folds I'm guessing. How would you use this in a production setting to predict on unseen data? Do you have to average the results?

**Comment by rohan-gt** [comment]
@toshihikoyanase Thanks! Couple of questions:
1.  The returned model outputs 3 sets of predictions for the 3 folds I'm guessing. How would you use this in a production setting to predict on unseen data? Do you have to average the results?
2. Does the folds parameter override the nfold parameter? Couldn't find anything in the documentation
3. The results don't seem to be reproducible even after setting `random_state` in LightGBM ans `seed` in LightGBMTunerCV

**Comment by toshihikoyanase** [comment]
@rohan-gt Thank you for aggregating the questions.

>1. The returned model outputs 3 sets of predictions for the 3 folds I'm guessing.

Yes, `CVBooster` returns multiple predictions. Actually, I haven't used the `LightGBM` in production systems, and I don't have answers. The developer of `return_cvbooster` mentioned that it can be used to enjoy ensemble techniques in https://github.com/microsoft/LightGBM/issues/283#issuecomment-642351645.

> req2. to do some ensemble techniques (stacking, averaging, etc) using the trained models from the cv() function

> 2. Does the folds parameter override the `nfold` parameter?

No, it doesn't. `LightGBMTuner` simply keeps the default value (5) or the value given by users. Does it cause problems?

> 3. The results don't seem to be reproducible even after setting random_state in LightGBM ans seed in LightGBMTunerCV.

Thank you for your feedback. I think this is because `LightGBMTuner` does not set the seed of Optuna samplers. We need to fix it.

> 4. When setting `num_boost_round=1000` and `early_stopping_rounds=100`, the `best_iteration` sometimes shows as 60.

Hmm, `LightGBMTuner` simply passes such arguments to `lightgbm.cv` function and I don't think it causes bugs.  But, anyway, I'll check it. 

> 5. Is there a way to use the default model as the baseline and improve from there?

No, it doesn't. The customize of `LightGBMTuner` is discussed in #1632, and I guess you can specify hyperparameter values using it. What do you think?

**Comment by rohan-gt** [comment]
@toshihikoyanase ,

2. > No, it doesn't. LightGBMTuner simply keeps the default value (5) or the value given by users. Does it cause problems?

Asked because you set `folds=KFold(n_splits=3)` and I was wondering if it overrides the default `nfold=5` parameter

4. > Hmm, LightGBMTuner simply passes such arguments to lightgbm.cv function and I don't think it causes bugs. But, anyway, I'll check it.

I got that result while testing it on the Titanic dataset

5. > No, it doesn't. The customize of LightGBMTuner is discussed in #1632, and I guess you can specify hyperparameter values using it. What do you think?

So the problem I'm facing is when I compare the performance of default LightGBM with no parameters to tuned LightGBM, the default model has a higher score. So I was thinking maybe keep the zeroth trial score from the default model score and start tuning the model score from that baseline
